block/manager.go

```diff
@@ -424,6 +427,15 @@ func (m *Manager) BatchRetrieveLoop(ctx context.Context) {
 				return
 			}
 
+			// If lastBatchHash is not set, retrieve the last batch hash from store
+			if m.lastBatchHash == nil {
```

Comment:
you don't need to do this in the BatchRetrieveLoop right? this can be done once in the NewManager?

Reply:
You are right, I can move this out 👌